File: Diagrammatize.Design.Pattern/code/cxx/Builder/A4/backup/TextBuilder.cpp
#include "TextBuilder.hpp"

TextBuilder::~TextBuilder() {
}

void TextBuilder::makeTitle(const std::string& title) {     // 纯文本的标题
    buffer.append("==============================\n");      // 装饰线
    buffer.append("『" + title + "』\n");                   // 为标题添加『』
    buffer.append("\n");                                    // 换行
}

void TextBuilder::makeString(const std::string& str) {      // 纯文本的字符串
    buffer.append("■" + str + "\n");                        // 为字符串添加■
    buffer.append("\n");                                    // 换行
}

void TextBuilder::makeItems(const std::vector<std::string>& items) {    // 纯文本的条目
    for (int i = 0; i < items.size(); i++) {
        buffer.append("　・" + items[i] + "\n");            // 为条目添加・
    }
    buffer.append("\n");                                    // 换行
}

void TextBuilder::close() {                             // 完成文档
    buffer.append("==============================\n");      // 装饰线
}

std::string TextBuilder::getResult() {                  // 完成的文档
    return buffer;
}

File: Diagrammatize.Design.Pattern/code/python/Builder/A4/TextBuilder.py
import io

from Builder import Builder

class TextBuilder(Builder):
    def __init__(self):
        self.buffer = ''       # 文档内容保存在该字段中 

    def makeTitle(self, title):                         # 纯文本的标题
        self.buffer += "==============================\n"       # 装饰线
        self.buffer += "『" + title + "』\n"                    # 为标题加上『』
        self.buffer += "\n"                                     # 换行

    def makeString(self, str_):                         # 纯文本的字符串
        self.buffer += "■" + str_ + "\n"                        # 为字符串添加■
        self.buffer += "\n"                                     # 换行

    def makeItems(self, items):                         # 纯文本的条目
        for i in range(len(items)):
            self.buffer += "　・" + items[i] + "\n"             # 为条目添加・
        self.buffer += "\n"                                     # 换行

    def close(self):                                    # 完成文档
        self.buffer += "==============================\n"       # 装饰线

    def getResult(self):                                # 完成后的文档
        return self.buffer
